parser/prism/Translator.h

```diff
@@ -12,6 +12,16 @@ extern "C" {
 
 namespace sorbet::parser::Prism {
 
+class TranslateResult final {
```

Comment:
I see why you had to do this, because `ParseResult` is defined in the (non-Prism) `Parser.h`. What if we extracted that type into its own header, so it could be reused here? Demo: #618

Alternatively, we could just replace that struct entirely with a simple tuple like  `std::tuple<std::vector<core::LocOffsets>, std::unique_ptr<parser::Node>>`?